presto-main/src/main/java/com/facebook/presto/sql/analyzer/MaterializedViewPlanValidator.java

```diff
@@ -57,17 +87,50 @@ protected Void visitJoin(Join node, MaterializedViewPlanValidatorContext context
             throw new SemanticException(NOT_SUPPORTED, node, "Only join-on is supported for materialized view.");
         }
 
+        process(node.getLeft(), context);
+        process(node.getRight(), context);
+
         context.setProcessingJoinNode(true);
         process(((JoinOn) joinCriteria).getExpression(), context);
         context.setProcessingJoinNode(false);
+
+        context.popJoinNode();
         return null;
     }
 
+    @Override
+    protected Void visitSetOperation(SetOperation node, MaterializedViewPlanValidatorContext context)
+    {
+        if (isValidationOnly) {
+            return super.visitSetOperation(node, context);
+        }
+
+        List<RelationType> outputDescriptorList = node.getRelations().stream()
```

Comment:
I would prefer to move the functionality to analyze base column mapping in a separate if check/method. 
That way we can have only one             return super.visitSetOperation(node, context);

Reply:
Revised and put the base column mapping population part into a separate method, so that there's only one call to super.visitComparisonExpression(node, context). Let me know if you meant differently.